aws/resource_aws_wafregional_xss_match_set.go

```diff
@@ -0,0 +1,173 @@
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsWafRegionalXssMatchSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafRegionalXssMatchSetCreate,
+		Read:   resourceAwsWafRegionalXssMatchSetRead,
+		Update: resourceAwsWafRegionalXssMatchSetUpdate,
+		Delete: resourceAwsWafRegionalXssMatchSetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"xss_match_tuples": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"field_to_match": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"data": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"type": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+						"text_transformation": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsWafRegionalXssMatchSetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafregionalconn
+	region := meta.(*AWSClient).region
+
+	log.Printf("[INFO] Creating XssMatchSet: %s", d.Get("name").(string))
+
+	wr := newWafRegionalRetryer(conn, region)
+	out, err := wr.RetryWithToken(func(token *string) (interface{}, error) {
+		params := &waf.CreateXssMatchSetInput{
+			ChangeToken: token,
+			Name:        aws.String(d.Get("name").(string)),
+		}
+
+		return conn.CreateXssMatchSet(params)
+	})
+	if err != nil {
+		return errwrap.Wrapf("[ERROR] Error creating XssMatchSet: {{err}}", err)
+	}
+	resp := out.(*waf.CreateXssMatchSetOutput)
+
+	d.SetId(*resp.XssMatchSet.XssMatchSetId)
+
+	return resourceAwsWafRegionalXssMatchSetUpdate(d, meta)
+}
+
+func resourceAwsWafRegionalXssMatchSetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafregionalconn
+	log.Printf("[INFO] Reading XssMatchSet: %s", d.Get("name").(string))
+	params := &waf.GetXssMatchSetInput{
+		XssMatchSetId: aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetXssMatchSet(params)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "WAFNonexistentItemException" {
+			log.Printf("[WARN] WAF IPSet (%s) not found, error code (404)", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return err
+	}
+
+	d.Set("name", resp.XssMatchSet.Name)
+
+	return nil
+}
+
+func resourceAwsWafRegionalXssMatchSetUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] Updating XssMatchSet: %s", d.Get("name").(string))
+	err := updateXssMatchSetResourceWR(d, meta, waf.ChangeActionInsert)
```

Comment:
I think this isn't right, because during update we may be both inserting and deleting tuples, not just inserting. See https://github.com/terraform-providers/terraform-provider-aws/commit/99e75ad4b733c8ee66e0d9e911f36e9c7ae2b062 for more context.